File: AdvancedMapAndroid/app/src/main/java/com/carto/advancedmap/sections/geocoding/adapter/GeocodingResultAdapter.java
package com.carto.advancedmap.sections.geocoding.adapter;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.view.View;
import android.view.ViewGroup;
import android.widget.AbsListView;
import android.widget.ArrayAdapter;

import com.carto.geocoding.GeocodingResult;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by aareundo on 21/08/2017.
 */

public class GeocodingResultAdapter extends ArrayAdapter<GeocodingResult> {

    List<GeocodingResult> items = new ArrayList<GeocodingResult>();
    public int width;

    public GeocodingResultAdapter(Context context) {
        super(context, -1);
    }

    @Override
    public int getCount() {
        return items.size();
    }

    @Nullable
    @Override
    public GeocodingResult getItem(int position) {
        return items.get(position);
    }

    @NonNull
    @Override
    public View getView(int position, @Nullable View convertView, @NonNull ViewGroup parent) {

        GeocodingResultCell cell;

        GeocodingResult item = items.get(position);

        if (convertView == null) {
            cell = new GeocodingResultCell(getContext());

            int height = (int)(40 * getContext().getResources().getDisplayMetrics().density);
            cell.setLayoutParams(new AbsListView.LayoutParams(width, height));
        } else {
            cell = (GeocodingResultCell)convertView;
        }

        cell.update(item);

        return cell;
    }
}

File: AdvancedMapKotlin/app/src/main/java/com/carto/advanced/kotlin/sections/geocoding/GeocodingResultAdapter.kt
package com.carto.advanced.kotlin.sections.geocoding

import android.content.Context
import android.view.View
import android.view.ViewGroup
import android.widget.AbsListView
import android.widget.ArrayAdapter
import com.carto.advanced.kotlin.sections.base.utils.CGRect
import com.carto.geocoding.GeocodingResult

/**
 * Created by aareundo on 20/07/2017.
 */
class GeocodingResultAdapter(context: Context?) : ArrayAdapter<GeocodingResult>(context, -1) {

    var items: MutableList<GeocodingResult> = mutableListOf()

    var width: Int = 0

    override fun getCount(): Int {
        return items.size
    }

    override fun getItem(position: Int): GeocodingResult {
        return items[position]
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup?): View {

        val cell: GeocodingResultCell?

        val item = items[position]

        if (convertView == null) {
            cell = GeocodingResultCell(context)

            val height = (40 * context.resources.displayMetrics.density).toInt()
            cell.layoutParams = AbsListView.LayoutParams(width, height)
            cell.frame = CGRect(0, 0, width, height)
        } else {
            cell = convertView as GeocodingResultCell
        }

        cell.update(item)
        cell.layoutSubviews()

        return cell
    }
}